src/error-catch-notifier.js

```diff
@@ -0,0 +1,235 @@
+    return args.split(',').map(function argsTrimCommentsAndWhiteSpace(arg) {
+    // Ensure no inline comments are parsed and trim the whitespace.
+        return arg.replace(/\/\*.*\*\//, '').trim();
+    }).filter(function filterUndefined(arg) {
+    // Ensure no undefined values are added.
+        return arg;
+    });
+}
+
+/**
+ * Determines if object is a function
+ * @param {object} obj - any value with object on its prototype chain
+ * @return {bool} true if obj is a function; false if not
+ */
+export function isFunction(obj) {
+    return !!(obj && obj.constructor && obj.call && obj.apply);
+}
+
+export function buildSubscriberList(errorSubscriberFunctions = []) {
+    return errorSubscriberFunctions.filter(function errorSubscriberFunctionsFilter(subscriberCallback, index) {
+        // subscribers must be a function
+        if (!isFunction(subscriberCallback)) {
+            if (_isLoggingEnabled) {
+                console.warn(`Skipping error subscriber: ${subscriberCallback} at errorSubscribers index ${index}`);
+                console.warn('Subscriber is not a function');
+            }
+
+            return false;
+        }
+
+        // subscribers must take an error as their first argument
+        const subscriberFunctionArgumentNames = getArgumentNames(subscriberCallback);
+        if (subscriberFunctionArgumentNames[0] !== 'error') {
+            if (_isLoggingEnabled) {
+                console.warn(`Skipping error subscriber: ${subscriberCallback.name}`);
+                console.warn('First argument of subscriber function must be named error');
+            }
+
+            return false;
+        }
+
+        return true;
+    });
+}
+
+/**
+ * Sets error messages to be logged to console
+ */
+export function enableLogging() {
+    _isLoggingEnabled = true;
+}
+
+export function disableLogging() {
+    _isLoggingEnabled = false;
+}
+
+/**
+ * Sets error catching on
+ */
+export function enableErrorCatching() {
+    if (_errorSubscribers.length === 0) {
+        if (_isLoggingEnabled) {
+            console.warn('No valid error subscribers provided. Use init to pass valid error subscribers');
+        }
+
+        _isEnabled = false;
+
+        return;
+    }
+
+    _isEnabled = true;
+}
+
+/**
+ * Sets error catching off
+ */
+export function disableErrorCatching() {
+    _isEnabled = false;
+
+    return true;
+}
+
+/**
+ * Sets _errorSubscribers array
+ * @param {function[]} errorSubscriberFunctions - array of ErrorSubscriber
+ * @param {ErrorSubscriber} errorSubscriberFunctions[] - function called when error is caught by wrap
+ * @param {bool} [enabled=false] - errors caught by wrap will be passed to errorSubscribers when true
+ * @param {bool} [loggingEnabled=false] - error messages will be logged to console when true
+ */
+export function initErrorCatchNotifier(errorSubscriberFunctions = [], enabled = false, loggingEnabled = false) {
+    loggingEnabled ? enableLogging() : disableLogging();
```

Comment:
I just use the functions I made for this in the init. Now reason to duplicate set behavior